### PR TITLE
feat(client): generate challenge data as static files

### PR DESCRIPTION
**Pull request by Nirajn2311**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

Ref - https://github.com/freeCodeCamp/freeCodeCamp/pull/50355#issuecomment-1544400365

Instead of relying on the generate page data by Gatsby, I'm generating the challenge data and storing them as individual files in the clients static folder so that mobile app can make use of them.

### CODE REVIEW

**Review by Sembauke — APPROVED** [review]

File: tools/scripts/build/build-external-curricula-data.ts [hunk 1]
```diff
@@ -71,24 +81,37 @@ export function buildExtCurriculumData(
...
         superBlock[superBlockKey]['blocks'][blockNames[j]]['challenges'] =
           curriculum[superBlockKey]['blocks'][blockNames[j]]['meta'];
+
+        const blockChallenges =
+          curriculum[superBlockKey]['blocks'][blockNames[j]]['challenges'];
+
+        for (let k = 0; k < blockChallenges.length; k++) {
+          const challenge = blockChallenges[k];
+          const challengeId = challenge['id'];
+          const challengePath = `challenges/${superBlockKey}/${blockNames[j]}/${challengeId}`;
```
Comment (naomi-lgbt) [inline tools/scripts/build/build-external-curricula-data.ts:111]:
```suggestion
          const challengePath = `challenges/${parse2022SuperBlock(
            superBlockKey
          )}/${blockNames[j]}/${challengeId}`;
```

File: tools/scripts/build/build-external-curricula-data.ts [hunk 2]
```diff
@@ -71,24 +81,37 @@ export function buildExtCurriculumData(
...
+
+          writeToFile(challengePath, challenge);
+        }
       }
 
       writeToFile(superBlockKey, superBlock);
```
Comment (naomi-lgbt) [inline tools/scripts/build/build-external-curricula-data.ts:117]: writeToFile(parse2022SuperBlock(superBlockKey), superBlock);

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
This isn't necessarily blocking, and if you can work with the weird directories that's fine.

But it looks like the `2022/responsive-web-design` is being parsed as a nested directory:

image

Review suggestions will fix this. Can't reach far enough down the file, but would also need to declare the function:

```js
  function parse2022SuperBlock(superBlockName: string) {
    return superBlockName.replace(/\//g, '-');
  }
```

**Comment by Nirajn2311** [comment]
> This isn't necessarily blocking, and if you can work with the weird directories that's fine.
> 
> But it looks like the `2022/responsive-web-design` is being parsed as a nested directory:

Yes, we're fine with this and actually I prefer it this way

**Review by naomi-lgbt — APPROVED** [review]
Works for me!

**Pull request merged**